common/src/main/java/com/wynntils/models/abilities/type/AbilityTreeInfo.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.abilities.type;
+
+import com.google.gson.JsonElement;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.text.PartStyle;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.type.IterationDecision;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+
+/**
+ * This class contains all relevant info to a specific class' ability tree.
+ */
+public class AbilityTreeInfo {
+    private static final File SAVE_FOLDER = WynntilsMod.getModStorageDir("debug");
```

Comment:
This class seem somewhat general, but then has parts that are highly specialized for dumping to a file. 

Is your intention to reuse this class later on to e.g. read the current players actual ability tree? If so, you might want to separate perhaps the stuff done for dumping to a json file, and the stuff needed when reading up the players actual tree.

Reply:
This class is what I use to represent a class' ability tree. Ideally only 5 instances of these ever exist, one per class. I have a `ParsedAbilityTree` which is a record of map of Ability Nodes - States. I plan to use that for storing "current" (modified, parsed, loaded, shared, etc) states. 

In fact, to answer your other comment. I realized that as well, and moved the relevant node parsing code to the node record itself, making the file much smaller, and the code fit in more. However, that did not end up being backported.

Reply:
@magicus Fine to merge this as is, and it is obviously going to be addressed in the next part?